src/test/java/io/opentracing/contrib/examples/activate_deactivate/Callback.java

```diff
@@ -15,27 +17,34 @@
 
   private static final Logger logger = LoggerFactory.getLogger(Callback.class);
 
-  private final Continuation continuation;
+  private final Random random = new Random();
 
-  Callback(ActiveSpan activeSpan) {
-    continuation = activeSpan.capture();
+  private final AutoFinishScope.Continuation continuation;
+
+  Callback(Scope activeSpan) {
+    continuation = ((AutoFinishScope)activeSpan).defer();
```

Comment:
Hey!

So, if you are talking about the chance of `Scope` not being a `AutoFinishScope` here, I can add the check for sanity purposes (even though we know that for this example, we always get an Auto one :) )

But if you are talking about doing such cast overall - that's a reason to try to include the `defer` and `Continuation` api into the main interface (i.e. `io.opentracing.Scope`)